{
    "body": "Running tests with an anaconda python was causing several tests to fail:\r\n- directory layout\r\n- flag handlers\r\n- package sanity\r\n- env\r\n- cmd/env\r\n- cmd/list\r\n- url\r\n- versions\r\n- lock\r\n\r\nReferencing #2738 since this is relevant\r\n\r\n```console\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\r\n\r\n    from spack import *\r\n\r\n\r\n>   class Callpath(CMakePackage):\r\n        \"\"\"Library for representing callpaths consistently in\r\nE       NameError: name 'CMakePackage' is not defined\r\n\r\nvar/spack/repos/builtin/packages/callpath/package.py:28: NameError\r\n_______________________________________________________________________ TestFlagHandlers.test_ld_flags_cmake _______________________________________________________________________\r\n\r\nself = <spack.test.flag_handlers.TestFlagHandlers object at 0x110f579b0>, temp_env = None\r\n\r\n    def test_ld_flags_cmake(self, temp_env):\r\n        s = spack.spec.Spec('callpath ldflags=-mthreads')\r\n>       s.concretize()\r\n\r\nlib/spack/spack/test/flag_handlers.py:162:\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\r\nlib/spack/spack/spec.py:1840: in concretize\r\n    user_spec_deps=user_spec_deps),\r\nlib/spack/spack/spec.py:2234: in normalize\r\n    self.validate_or_raise()\r\nlib/spack/spack/spec.py:2286: in validate_or_raise\r\n    spack.repo.get(spec.fullname)\r\nlib/spack/spack/repo.py:1120: in get\r\n    return path.get(spec)\r\nlib/spack/spack/repo.py:101: in converter\r\n    return function(self, spec_like, *args, **kwargs)\r\nlib/spack/spack/repo.py:577: in get\r\n    return self.repo_for_pkg(spec).get(spec)\r\nlib/spack/spack/repo.py:101: in converter\r\n    return function(self, spec_like, *args, **kwargs)\r\nlib/spack/spack/repo.py:831: in get\r\n    package_class = self.get_pkg_class(spec.name)\r\nlib/spack/spack/repo.py:1020: in get_pkg_class\r\n    module = self._get_pkg_module(pkg_name)\r\nlib/spack/spack/repo.py:994: in _get_pkg_module\r\n    module = imp.load_source(fullname, file_path, f)\r\n/anaconda3/lib/python3.6/imp.py:172: in load_source\r\n    module = _load(spec)\r\n<frozen importlib._bootstrap>:684: in _load\r\n    ???\r\n<frozen importlib._bootstrap>:665: in _load_unlocked\r\n    ???\r\n<frozen importlib._bootstrap_external>:678: in exec_module\r\n    ???\r\n<frozen importlib._bootstrap>:219: in _call_with_frames_removed\r\n    ???\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\r\n\r\n    from spack import *\r\n\r\n\r\n>   class Callpath(CMakePackage):\r\n        \"\"\"Library for representing callpaths consistently in\r\nE       NameError: name 'CMakePackage' is not defined\r\n\r\nvar/spack/repos/builtin/packages/callpath/package.py:28: NameError\r\n_______________________________________________________________________ TestFlagHandlers.test_ld_libs_cmake ________________________________________________________________________\r\n\r\nself = <spack.test.flag_handlers.TestFlagHandlers object at 0x1135ee978>, temp_env = None\r\n\r\n    def test_ld_libs_cmake(self, temp_env):\r\n        s = spack.spec.Spec('callpath ldlibs=-lfoo')\r\n>       s.concretize()\r\n\r\nlib/spack/spack/test/flag_handlers.py:178:\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\r\nlib/spack/spack/spec.py:1840: in concretize\r\n    user_spec_deps=user_spec_deps),\r\nlib/spack/spack/spec.py:2234: in normalize\r\n    self.validate_or_raise()\r\nlib/spack/spack/spec.py:2286: in validate_or_raise\r\n    spack.repo.get(spec.fullname)\r\nlib/spack/spack/repo.py:1120: in get\r\n    return path.get(spec)\r\nlib/spack/spack/repo.py:101: in converter\r\n    return function(self, spec_like, *args, **kwargs)\r\nlib/spack/spack/repo.py:577: in get\r\n    return self.repo_for_pkg(spec).get(spec)\r\nlib/spack/spack/repo.py:101: in converter\r\n    return function(self, spec_like, *args, **kwargs)\r\nlib/spack/spack/repo.py:831: in get\r\n    package_class = self.get_pkg_class(spec.name)\r\nlib/spack/spack/repo.py:1020: in get_pkg_class\r\n    module = self._get_pkg_module(pkg_name)\r\nlib/spack/spack/repo.py:994: in _get_pkg_module\r\n    module = imp.load_source(fullname, file_path, f)\r\n/anaconda3/lib/python3.6/imp.py:172: in load_source\r\n    module = _load(spec)\r\n<frozen importlib._bootstrap>:684: in _load\r\n    ???\r\n<frozen importlib._bootstrap>:665: in _load_unlocked\r\n    ???\r\n<frozen importlib._bootstrap_external>:678: in exec_module\r\n    ???\r\n<frozen importlib._bootstrap>:219: in _call_with_frames_removed\r\n    ???\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\r\n\r\n    from spack import *\r\n\r\n\r\n>   class Callpath(CMakePackage):\r\n        \"\"\"Library for representing callpaths consistently in\r\nE       NameError: name 'CMakePackage' is not defined\r\n```\r\n### Expected Result\r\n\r\nAll the tests mentioned above to pass\r\n\r\n### Actual Result\r\n\r\nTests failing in a similar manner as pasted in the log output above.\r\n\r\n### Steps to reproduce the issue\r\nInstall Anaconda Python\r\n`export PATH=/anaconda3/bin:$PATH` in ~/.bashrc\r\nspack test flag_handler\r\n\r\n### Information on your system . \r\nmacOS High Sierra \r\n\r\n\r\nAnyone else experienced this error before?",
    "user": "mamelara",
    "url": "https://api.github.com/repos/spack/spack/issues/8413",
    "updated_at": "2018-06-06 21:35:08",
    "created_at": "2018-06-06 21:04:27",
    "closed_at": "2018-06-06 21:35:08",
    "state": "closed",
    "title": "Anaconda Python causes tests to fail - import errors?",
    "number": 8413,
    "milestone": null,
    "labels": [
        "duplicate",
        "python3"
    ],
    "id": 330032208,
    "html_url": "https://github.com/spack/spack/issues/8413",
    "assignees": [],
    "comments": 2
}